lib/storage/metadata/mongoclient/LogConsumer.js

```diff
@@ -151,15 +66,28 @@ class LogConsumer {
      */
     readRecords(params, cb) {
         const limit = params.limit || 10000;
-        const startIDandSeq = params.startSeq.toString().split('|');
-        const startSeq = parseInt(startIDandSeq[0], 10) || 0;
-        const lastEndID = startIDandSeq[1];
-        const recordStream = new ListRecordStream(this.logger, lastEndID);
+        let startSeq;
+        if (params.startSeq) {
+            try {
+                // parse the opaque JSON string passed through from a
+                // previous 'info' event
+                startSeq = JSON.parse(params.startSeq);
+            } catch (err) {
+                this.logger.error('malformed startSeq', {
+                    startSeq: params.startSeq,
+                });
+                // start over
+                startSeq = { ts: 0 };
+            }
+        } else {
```

Comment:
Minor: we can avoid the else block here if we initially set `startSeq` to `{ ts: 0 }`.

Reply:
done